RIOT-OS/RIOT
# net/rdcli*: restructure and rename RD related modules

### Contribution description
As discussed in #7428: the Resource Directory endpoint implementations that we merged (i.e. `rdcli` and `rdcli_simple`) use some naming conventions that are not in line with the wording used in [draft-ietf-core-resource-directory-15](https://tools.ietf.org/html/draft-ietf-core-resource-directory-15).

This PR restructures all existing resource directory code by applying a more clear, and extensible structure, while following the wording used in the draft.

I introduced a high-level module called `cord`, which includes all the code related to interacting with resource directories. All existing (endpoint) and future (lookup client) code is put as submodules into this high level module. See `sys/net/application_layer/cord/doc.txt` as included in this PR for more information on the high-level structure.

Sorry for the amount of lines of code in this PR, but I think it does make sense to have all these changes in a single PR. I tried to do the changes in self-contained commits to make reviewing as easy as possible...

### Testing procedure
Simply verify that the two example applications (now `cord_ep` and `cord_epsim`) are still working as expected.

Also check the generated doxygen.

### Issues/PRs references
#7428
